codegen/src/api/mod.rs

```diff
@@ -551,7 +496,8 @@ impl RuntimeGenerator {
                 /// The error type returned when there is a runtime issue.
                 pub type DispatchError = #types_mod_ident::sp_runtime::DispatchError;
 
-                #outer_event
+                /// The outer event enum.
+                pub type Event = #event_path;
 
                 impl #crate_path::events::RootEvent for Event {
```

Comment:
Just to add another simplification; `RootEvent`, `RootExtrinsic` and `RootError` traits can all be removed now, since we now have type IDs and thus are able to decode the call/error/tx into whatever type implementing `DecodeAsType` makes sense. This should simplify the code a bunch more :)